vms/platformvm/warp/validator.go

```diff
@@ -32,6 +32,11 @@ type ValidatorState interface {
 	GetValidatorSet(ctx context.Context, height uint64, subnetID ids.ID) (map[ids.NodeID]*validators.GetValidatorOutput, error)
 }
 
+type CanonicalValidatorSet struct {
+	Validators  []*Validator
+	TotalWeight uint64
+}
```

Comment:
I dislike how we are opening up the ability for the user of this API to provide incorrect weights...

`FlattenValidatorSet` only returned the total weight to avoid needing to use safemath logic for creating the weight of the validators... It seems like a cleaner API if we removed the total weight from `FlattenValidatorSet` and calculate it later inside of `Verify`.

Alternatively, we could unexport `Validators` and `TotalWeight` so that we enforce the set is correctly calculated by this package (but that seems less ideal)

Reply:
I think that I'd like to let the caller make the decision here, and like the idea of calculating the total weight inside the verify function.